src/lib/tpm.c

```diff
@@ -486,6 +499,77 @@ bool files_load_bytes_from_path(const char *path, UINT8 *buf, UINT16 *size) {
 LOAD_TYPE(TPM2B_PUBLIC, public)
 LOAD_TYPE(TPM2B_PRIVATE, private)
 
+CK_RV tpm_get_token_info_from_tpm (tpm_ctx *ctx, CK_TOKEN_INFO *info) {
+
+    uint32_t revision;
+    uint32_t version;
+    uint32_t manufacturer;
+    unsigned char manufacturerID[5] = {0}; // 4 bytes + '\0'
+    uint32_t vendor[4];
+    CK_RV rv = CKR_OK;
+    TPM2_CAP capability = TPM2_CAP_TPM_PROPERTIES;
+    UINT32 property = TPM2_PT_FIXED;
+    UINT32 propertyCount = TPM2_MAX_TPM_PROPERTIES;
+    TPMS_CAPABILITY_DATA *capabilityData;
+    TPMI_YES_NO moreData;
+
+    check_pointer(ctx);
+    check_pointer(info);
+
+    TSS2_RC rval = Esys_GetCapability(ctx->esys_ctx,
+        ESYS_TR_NONE,
+        ESYS_TR_NONE,
+        ESYS_TR_NONE,
+        capability,
+        property, propertyCount, &moreData, &capabilityData);
+
+    if (rval != TSS2_RC_SUCCESS) {
+        LOGE("Esys_GetCapability: 0x%x:", rval);
+        return CKR_GENERAL_ERROR;
+    }
+
+    if (!capabilityData || capabilityData->data.tpmProperties.count < TPM2_PT_VENDOR_STRING_4 - TPM2_PT_FIXED + 1) {
```

Comment:
some of these seem overlength....

Reply:
in terms of line lenght - yes.
I'll split them into two lines if it improves readability.
Coming from kernel space, I know how code sometimes looks if the line length is strictly enforced...
Maybe it would make sense to document somewhere what our coding style should look like (over all tpm2-software projects).